libs/form/src/next/upload/vl-upload.component.ts

```diff
@@ -0,0 +1,586 @@
+        super.firstUpdated(changedProperties);
+
+        this.initializeComponent();
+    }
+
+    render(): TemplateResult {
+        const uploadClasses = {
+            'vl-upload': true,
+            'vl-upload--disabled': this.disabled,
+            'vl-upload--error': this.isInvalid || this.error,
+            'vl-upload--success': this.success,
+        };
+
+        const dropzoneContainerClasses = {
+            'dz-clickable': this.readonly,
+            dropzone: true,
+            'vl-upload__element__label': true,
+        };
+
+        return html`
+            ${this.getUploadElementTemplate()} ${this.getPreviewTemplate()}
+            <div class="vl-upload__files">
+                <div class="vl-upload__files__container"></div>
+                <div class="vl-upload__files__input__container"></div>
+                <button class="vl-upload__files__close vl-link vl-link--icon">
+                    <span class="vl-link__icon vl-vi vl-vi-trash" aria-hidden="true"></span>
+                    Verwijder alle bestanden
+                </button>
+            </div>
+            <div class=${classMap(uploadClasses)} aria-label=${this.label}>
+                <div class="vl-upload__element">
+                    <div class="vl-upload__overlay">
+                        <p class="vl-upload__overlay__text">
+                            <span class="vl-link__icon vl-vi vl-vi-paperclip" aria-hidden="true"></span>
+                        </p>
+                    </div>
+                    <div
+                        id="dropzone-container"
+                        class=${classMap(dropzoneContainerClasses)}
+                        aria-label=${this.label}
+                    ></div>
+                </div>
+            </div>
+        `;
+    }
+
+    disconnectedCallback() {
+        super.disconnectedCallback();
+
+        this.removeDropzoneEvents();
+        if (this.dropzoneInstance) {
+            this.updateFileList(this.dropzoneInstance);
+            this.dropzoneInstance.destroy();
+        }
+    }
+
+    updated(changedProperties: Map<string, unknown>) {
+        super.updated(changedProperties);
+
+        if (changedProperties.has('value')) {
+            this.setValue(this.value ?? '');
+        }
+
+        if (changedProperties.has('disabled')) {
+            this.disabled ? this.dropzoneInstance?.disable() : this.dropzoneInstance?.enable();
+
+            if (this.disabled) {
+                this.dropzoneInstance?.hiddenFileInput?.setAttribute('disabled', 'disabled');
+                this.getUploadButton()?.setAttribute('disabled', 'disabled');
+                this.dropzoneInstance?.disable();
+            } else {
+                this.dropzoneInstance?.hiddenFileInput?.removeAttribute('disabled');
+                this.getUploadButton()?.removeAttribute('disabled');
+                this.dropzoneInstance?.enable();
+            }
+            if (this.getInput()) this.getInput()!.disabled = this.disabled;
```

Comment:
Het `!` is niet nodig als je al checkt of het bestaat in het if statement, ook liefst maar 1x de this.getInput() aanroepen en deze in een var stoppen.

Reply:
Ja ik weet het, maar voor of andere reden krijg ik wel onderstaande in de IDE; echter als ik `this.getInput()` in variabele stop zoals je adviseert, dan werkt het wel zoals verwacht.

<img width="855" alt="image" src="https://github.com/milieuinfo/uigov-web-components/assets/15731348/4aae4582-e8bc-4000-ae0a-13360e1a4600">

Pas het aan